internal/xds/translator/route.go

```diff
@@ -214,6 +216,36 @@ func buildXdsRedirectAction(redirection *ir.Redirect) *route.RedirectAction {
 	return ret
 }
 
+func buildXdsURLRewriteAction(rou *route.Route, routeName string, urlRewrite *ir.URLRewrite) *route.RouteAction {
+	ret := &route.RouteAction{
+		ClusterSpecifier: &route.RouteAction_Cluster{
+			Cluster: routeName,
+		},
+	}
+
+	if urlRewrite.Path != nil {
+		if urlRewrite.Path.FullReplace != nil {
+			ret.RegexRewrite = &matcher.RegexMatchAndSubstitute{
+				Pattern: &matcher.RegexMatcher{
+					Regex: "/.+",
+				},
+				Substitution: *urlRewrite.Path.FullReplace,
+			}
+		} else if urlRewrite.Path.PrefixMatchReplace != nil {
+			ret.PrefixRewrite = *urlRewrite.Path.PrefixMatchReplace
+		}
+	}
+
+	if urlRewrite.Hostname != nil {
+		ret.HostRewriteSpecifier = &route.RouteAction_HostRewriteLiteral{
+			HostRewriteLiteral: *urlRewrite.Hostname}
```

Comment:
curious why you picked this over `host_rewrite_header` https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-regex-rewrite
dont they seem to be doing the same thing ?

Reply:
They are different actually:

host_rewrite_header

([string](https://developers.google.com/protocol-buffers/docs/proto#scalar)) Indicates that during forwarding, the host header will be swapped with the content of given downstream or [custom](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#config-http-conn-man-headers-custom-request-headers) header. If header value is empty, host header is left intact. Using this option will append the [x-forwarded-host](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#config-http-conn-man-headers-x-forwarded-host) header if [append_x_forwarded_host](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-append-x-forwarded-host) is set.

host_rewrite_literal

([string](https://developers.google.com/protocol-buffers/docs/proto#scalar)) Indicates that during forwarding, the host header will be swapped with this value. Using this option will append the [x-forwarded-host](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#config-http-conn-man-headers-x-forwarded-host) header if [append_x_forwarded_host](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-append-x-forwarded-host) is set.

Reply:
241 "}" to the new line?

Reply:
You mean the code style? @qicz

Reply:
@arkodg I have refactored the gateway-api translator, lines of translator reduce from 2000 to 150. Codes of gateway-api translator are more organized, easy-to-extend, reading-friendly now.